code/espurna/config/hardware.h

```diff
@@ -4212,6 +4212,27 @@
     #define SENSOR_POWER_UNITS          POWER_WATTS
 
 // -----------------------------------------------------------------------------
+// KINGART_CURTAIN_SWITCH
+// -----------------------------------------------------------------------------
+#elif defined(KINGART_CURTAIN_SWITCH)
+
+    // Info
+    #define MANUFACTURER            "KingArt"
+    #define DEVICE                  "KINGART_CURTAIN_SWITCH"
+    #define KA_CURTAIN_SUPPORT      1
+
+    #define KA_CURTAIN_HW_PORT      Serial      // Hardware serial port (if UART_MQTT_USE_SOFT == 0)
+    #define KA_CURTAIN_BAUDRATE     19200      // Serial speed
+    #define KA_CURTAIN_TERMINATION  '\e'         // Termination character
+    #define KA_CURTAIN_BUFFER_SIZE  100         // UART buffer size
```

Comment:
Move those inside of .ino?
I cant add build test `#define KA_CURTAIN_SUPPORT 1` into the `code/test/nondefault.h` otherwise.
In case any other device needs the same module for some reason, defines can be surrounded with `#ifndef \n #define ... default \n #endif`

Reply:
Changed, but also working on a KINGART dimmer so I might change the filename of the ino later and make it a combined KINGART module.